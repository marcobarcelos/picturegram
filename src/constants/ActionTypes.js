export const SHOTS_REQUEST = 'SHOTS_REQUEST';
export const SHOTS_RECEIVED = 'SHOTS_RECEIVED';
export const SHOTS_ERROR = 'SHOTS_ERROR';
