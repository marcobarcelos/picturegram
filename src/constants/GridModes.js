export const GRID_MODE_LARGE = 'GRID_MODE_LARGE';
export const GRID_MODE_SMALL = 'GRID_MODE_SMALL';
