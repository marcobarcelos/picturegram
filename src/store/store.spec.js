import * as types from '../constants/ActionTypes';
import { configureStore } from './configureStore';
import { GRID_MODE_SMALL, GRID_MODE_LARGE } from '../constants/GridModes';

describe('Store', () => {
  it('should handle actions', () => {
    const store = configureStore();
    const actions = [
      { type: types.GRID_MODE_UPDATE, gridMode: GRID_MODE_SMALL },
      { type: types.SHOT_DETAIL_SELECT, selectedShot: { id: 1 } },
      { type: types.GRID_MODE_UPDATE, gridMode: GRID_MODE_LARGE },
      { type: types.SHOT_DETAIL_SELECT, selectedShot: { id: 2 } },
      { type: types.GRID_MODE_UPDATE, gridMode: GRID_MODE_SMALL },
      { type: types.SHOTS_REPLACE, shots: [{ id: 1 }, { id: 2 }, { id: 3 }] },
      { type: types.SHOTS_REPLACE, shots: [{ id: 4 }, { id: 5 }, { id: 6 }] },
      { type: types.SHOTS_APPEND, shots: [{ id: 7 }, { id: 8 }] },
      { type: types.FETCH_REQUEST },
      { type: types.FETCH_ERROR }
    ];

    actions.forEach(action => store.dispatch(action));

    const actual = store.getState();
    expect(actual.shots).toMatchSnapshot();
  });
});
