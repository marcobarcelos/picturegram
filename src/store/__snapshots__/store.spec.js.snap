// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Store should handle actions 1`] = `
Object {
  "error": "An error occurred while fetching shots",
  "gridMode": "GRID_MODE_SMALL",
  "items": Array [
    Object {
      "id": 4,
    },
    Object {
      "id": 5,
    },
    Object {
      "id": 6,
    },
    Object {
      "id": 7,
    },
    Object {
      "id": 8,
    },
  ],
  "links": Object {
    "next": undefined,
  },
  "loading": false,
  "selectedItem": Object {
    "id": 2,
  },
}
`;
